### DISCUSSION TITLE
Show connection status in notification icon

### DISCUSSION THREAD

**Discussion by I-I-IT** [mullvad/mullvadvpn-app#8929] [Feature requests & Ideas]
### I have checked if others have suggested this already

- [X] I have checked this issue tracker to see if others have reported similar issues.

### Feature description

Currently, the Mullvad app displays the mullvad logo in the notification tray, regardless of the connection status. Other apps, like Proton VPN do it in a better way. When it is connecting, the Proton VPN app logo will still display but there will be a loading icon on the top right, when connected a check., etc.

My proposal will be to have a connection status icon in the notification status bar, like it is on Desktop.

### Alternative solutions

Pop-up notifications could also work, but I prefer what I recommended above.

### Type of feature

- [ ] Better privacy/anonymity
- [ ] Better at circumventing censorship
- [X] Easier to use
- [ ] Other

### Operating System

- [X] Android
- [ ] iOS
- [ ] Windows
- [ ] macOS
- [ ] Linux

**Comment by MrChocolatine** [comment]
I would like that as well.

**Comment by Pururun** [comment]
Thanks for the feedback. This does indeed sounds like something useful, I have created an internal issue for it.

**Comment by Morfyuum** [comment]
Hi, just wanted to add a concrete data point to this request.

I ran into this on Android (Pixel 10 Pro, GrapheneOS, app version 2026.8) and initially thought the icon change was actually broken, so I dug into it with `adb shell dumpsys notification`. Turns out the notification title text does update correctly ("Verbindung getrennt" → "Verbunden" / "Disconnected" → "Connected"), but the icon resource itself is identical in both states (`icon=Icon(typ=RESOURCE pkg=net.mullvad.mullvadvpn id=0x7f0700af)` in both the disconnected and the confirmed-connected, validated-tunnel state). So it's not a rendering glitch or caching issue — the app posts the same static icon regardless of tunnel state.

I filed this as #10887, which got closed as a duplicate of this discussion — makes sense, since a full status-icon system (like the desktop app / ProtonVPN's loading/checkmark approach) would obviously cover this too. Just wanted to leave the technical detail here in case it's useful when this gets picked up, since even the minimal version (distinct icon for connected vs. disconnected, without a full loading-state system) would already fix the main pain point: right now you can't tell tunnel state from the status bar at a glance, only by pulling down the notification and reading the text.

Thanks for already having an internal issue tracking this since 2024!

**Reply by Pururun** [reply]
Thanks for showing interest into this, I have bumped this a bit in priority, so it might happen sometime in the not too far future.
